### ISSUE TITLE
JavaScript Error in OSX app 10.12 Sierra

### ISSUE THREAD

**Issue by nickooolas** [salomvary/soundcleod#130]
Hey there,

Getting this error when trying to load up SoundCleod

**A JavaScript error occurred in the main process**

```Uncaught Exception:
TypeError: Error processing argument at index 0, conversion failure from #<Object>
    at TypeError (native)
    at validateState (/Applications/SoundCleod.app/Contents/Resources/app.asar/node_modules/electron-window-state/index.js:46:34)
    at module.exports (/Applications/SoundCleod.app/Contents/Resources/app.asar/node_modules/electron-window-state/index.js:154:3)
    at App.<anonymous> (/Applications/SoundCleod.app/Contents/Resources/app.asar/main.js:67:27)
    at emitTwo (events.js:111:20)
    at App.emit (events.js:191:7)
```

Could there be any environment things affecting it's operation?

If you need any more information, state, logs, let me know what you need and I'll get it.

Cheers,

Nick

**Comment by nickooolas** [comment]
Should have done so before, but a fresh install fixed the problem, don't know if this is specific to my env's?

Will leave it open for you to close if it's worthwhile for reference.

**Comment by salomvary** [comment]
Thanks, have not seen this before. Do you happen to know what version you had before the fresh install?

**Comment by nickooolas** [comment]
It was 1.1.6. Not sure what happened before it started stuffing up, but can confirm the fresh install fixed it :ok_hand:

**Comment by nickooolas** [comment]
Getting this one again. Any info I can provide to help out? @salomvary

**Comment by salomvary** [comment]
@nickooolas looks like it has to do something with the module that does the remembering of the window position and size but I still have little ideas what might be it. Do you use a macBook with external screen or some other multi-screen setup?

When SoundCleod does this again, can you try sharing the contents of the following file: `~/Library/Application Support/SoundCleod/window-state.json` (Open Finder, shift+cmd+g, paste `~/Library/Application Support/SoundCleod`)

**Comment by nickooolas** [comment]
Hey, I do indeed, I often switch between (two external, clamshell mode for MBP) at work, and then at home normally on it's own (internal screen, or one other Thunderbolt Display).

`{"width":1024,"height":640,"x":null,"y":178,"isMaximized":true,"isFullScreen":false,"displayBounds":{"x":0,"y":0,"width":1440,"height":900}}
`

**Comment by salomvary** [comment]
Thanks, that's useful information, I will investigate further.

**Comment by nickooolas** [comment]
Nice one, thanks @salomvary)

**Comment by salomvary** [comment]
Just released the fix in 1.1.7. \o/ Thanks for the patience, let me know if it still crashes.
